### PR TITLE
updated docs for agents and chatbots

### PR DESCRIPTION
**Pull request by martyna-mindsdb**
## Description

updated docs for agents and chatbots

related PR - https://github.com/mindsdb/mindsdb/pull/9481

Fixes #issue_number

## Type of change

- [ ] 📄 This change is a documentation update

### CODE REVIEW

File: docs/mindsdb_sql/agents/agent.mdx [hunk 1]
```diff
@@ -96,6 +96,19 @@ An agent can be created, deleted, queried, and updated. Here is how you can do t
        skills = ['test_skill']; -- this must be created with CREATE SKILL
     ```
 
+    Alternatively, you can create the model at the agent creation time based on [the model providers defined here](/integrations/ai-engines/langchain#openai).
```
Comment (ea-rus) [inline docs/mindsdb_sql/agents/agent.mdx:99]: Here is the model is not created, only an angent. 
How it works: mindsdb agent initializes langchain with llm settings (provider, api_key, model). Creation of model in mindsdb is not required. 
The model in mindsdb is used when it exists (and provider is mindsdb)

File: docs/mindsdb_sql/agents/agent.mdx [hunk 2]
```diff
@@ -96,6 +96,19 @@ An agent can be created, deleted, queried, and updated. Here is how you can do t
...
+
+    ```sql
+    CREATE AGENT my_agent
+    USING 
+        skills=['text_to_sql_skill'], -- this must be created with CREATE SKILL
+        provider='openai', -- choose one of the available model providers (openai, anthropic, anyscale, ollama, litellm, mindsdb)
+        model_name='gpt-4', -- define the model from the provider
```
Comment (ea-rus) [inline docs/mindsdb_sql/agents/agent.mdx:106]: this param was renamed to 'model' in https://github.com/mindsdb/mindsdb/pull/9525
depending on provider it takes it as mindsb model or llm model. 
if provider is not set it tries to detect from model name

**Review by ea-rus — CHANGES_REQUESTED** [review]

File: docs/mindsdb_sql/agents/agent.mdx (hunk 2 above)
Comment (martyna-mindsdb) [reply docs/mindsdb_sql/agents/agent.mdx:106]: @ea-rus 
I looked at the PR you linked, but it doesn't seem like it is renamed to `model`. For example, in [this line](https://github.com/mindsdb/mindsdb/pull/9525/files#diff-8e3c9fbf160b6c98edf9ddb1b4d69f0e2b48f6fc1d6ff39b6e7f481a8daa7f4eL32) there is still `model_name` used.

File: docs/mindsdb_sql/agents/agent.mdx (hunk 1 above)
Comment (martyna-mindsdb) [reply docs/mindsdb_sql/agents/agent.mdx:99]: I'll change it to the below:
```
Alternatively, you can create an agent and define the model to be used by an agent at the agent creation time based on [the model providers defined here](/integrations/ai-engines/langchain#openai).
```

File: docs/mindsdb_sql/agents/agent.mdx (hunk 2 above)
Comment (ea-rus) [reply docs/mindsdb_sql/agents/agent.mdx:106]: [this code](https://github.com/mindsdb/mindsdb/pull/9525/files#diff-8e3c9fbf160b6c98edf9ddb1b4d69f0e2b48f6fc1d6ff39b6e7f481a8daa7f4eL32) is from http api endpoint. there is model_name used
for  sql interface this is the line where it is used:
[this line](https://github.com/mindsdb/mindsdb/pull/9525/files#diff-6b2ec22184b7bc7f52fe2e03553a7bf35f9e384877b2d529c5b0e5844703c9a9R1440)

Comment (martyna-mindsdb) [reply docs/mindsdb_sql/agents/agent.mdx:106]: Ok. I updated it to `model`.

But this param name is for the CREATE AGENT syntax, and LangChain handler still uses `model_name`.

**Review by ea-rus — APPROVED** [review]

**Pull request merged**